### PR TITLE
Pass a pollset_set to some client handshakers.

### PR DESCRIPTION
**Pull request by daniel-j-born**

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @daniel-j-born isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
The Linux Foundation CLA GitHub bot_

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +0.9%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +0.9%
      +2.1%     +10 on_handshake_done                                            +10  +2.1%
      +1.6%      +8 connected                                                     +8  +1.6%
  +0.9%     +16 src/core/lib/channel/handshaker.cc                           +16  +0.9%
      +3.7%     +16 grpc_handshake_manager_do_handshake                          +16  +3.7%
  +1.4%     +16 src/core/lib/http/httpcli_security_connector.cc              +16  +1.4%
      +3.2%     +16 ssl_handshake                                                +16  +3.2%
      +0.7%      +1 on_handshake_done                                             +1  +0.7%

 -+-+-+-+-+-+-+ MIXED                                                    +-+-+-+-+-+-+-
  +0.0%     +24 [None]                                                      -128  -0.0%

  +0.0%     +72 TOTAL                                                        -80  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by daniel-j-born** [comment]
I signed it

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +0.9%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +0.9%
      +2.1%     +10 on_handshake_done                                            +10  +2.1%
      +1.6%      +8 connected                                                     +8  +1.6%
  +0.9%     +16 src/core/lib/channel/handshaker.cc                           +16  +0.9%
      +3.7%     +16 grpc_handshake_manager_do_handshake                          +16  +3.7%
  +1.4%     +16 src/core/lib/http/httpcli_security_connector.cc              +16  +1.4%
      +3.2%     +16 ssl_handshake                                                +16  +3.2%
      +0.7%      +1 on_handshake_done                                             +1  +0.7%

 -+-+-+-+-+-+-+ MIXED                                                    +-+-+-+-+-+-+-
  +0.0%     +24 [None]                                                      -128  -0.0%

  +0.0%     +72 TOTAL                                                        -80  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/lib/channel/handshaker.h [hunk 1]
```diff
@@ -135,11 +136,13 @@ void grpc_handshake_manager_shutdown(grpc_exec_ctx* exec_ctx,
                                      grpc_error* why);
 
 /// Invokes handshakers in the order they were added.
+/// \a interested_parties may be non-nullptr to provide a pollset that
```
Comment (markdroth) [inline src/core/lib/channel/handshaker.h]: s/pollset/pollset_set/

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +1.4%     +24 src/core/lib/channel/handshaker.cc                           +24  +1.4%
      +5.6%     +24 grpc_handshake_manager_do_handshake                          +24  +5.6%
  +0.9%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +0.9%
      +3.0%     +16 connected                                                    +16  +3.0%
      +0.4%      +2 on_handshake_done                                             +2  +0.4%
  +1.4%     +16 src/core/lib/http/httpcli_security_connector.cc              +16  +1.4%
      +3.2%     +16 ssl_handshake                                                +16  +3.2%
      +0.7%      +1 on_handshake_done                                             +1  +0.7%

 -------------- SHRINKING                                                --------------
  -0.0%      -8 [None]                                                       -88  -0.0%
  -0.4%      -8 src/core/ext/transport/chttp2/server/chttp2_server.cc         -8  -0.4%
      -0.8%      -8 grpc_chttp2_server_add_port                                   -8  -0.8%
      -5.1%      -2 [Unmapped]                                                    -2  -5.1%

  +0.0%     +40 TOTAL                                                        -40  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by daniel-j-born** [comment]
A required test is flaky and blocking merge: "Portability Tests Linux [Build Only] (internal CI)"

**Comment by markdroth** [comment]
Looks like it was a timeout.  I've asked Kokoro to rerun it.

Please tag known issues for any other failures before merging.

**Comment by daniel-j-born** [comment]
How do I find known issues and how do I tag them?

**Comment by markdroth** [comment]
Known issues: #12917 #13426 #13446 #13404 #13460 #12510 #13445 

Please squash your commits before merging.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +1.4%     +24 src/core/lib/channel/handshaker.cc                           +24  +1.4%
      +5.6%     +24 grpc_handshake_manager_do_handshake                          +24  +5.6%
  +0.9%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +0.9%
      +3.0%     +16 connected                                                    +16  +3.0%
      +0.4%      +2 on_handshake_done                                             +2  +0.4%
  +1.4%     +16 src/core/lib/http/httpcli_security_connector.cc              +16  +1.4%
      +3.2%     +16 ssl_handshake                                                +16  +3.2%
      +0.7%      +1 on_handshake_done                                             +1  +0.7%

 -------------- SHRINKING                                                --------------
  -0.4%      -8 src/core/ext/transport/chttp2/server/chttp2_server.cc         -8  -0.4%
      -0.8%      -8 grpc_chttp2_server_add_port                                   -8  -0.8%
      -5.1%      -2 [Unmapped]                                                    -2  -5.1%

 -+-+-+-+-+-+-+ MIXED                                                    +-+-+-+-+-+-+-
  +0.0%     +24 [None]                                                      -104  -0.0%

  +0.0%     +72 TOTAL                                                        -56  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by daniel-j-born** [comment]
The same required test is failing, so I can't merge ("Portability Tests Linux [Build Only] (internal CI)").

How do I trigger a rebuild in Kokoro?

Can you just disable all the merge requirements until you get them fixed?

**Comment by markdroth** [comment]
I've asked kokoro to rebuild the required test.

It's unusual for this particular test to be flaky, and the check is important, which is why it's required.

**Comment by daniel-j-born** [comment]
Still fails.

Log entries of note:
"Internal CI infrastructure error. Please ignore for now."

It looks like the tests pass ("PASSED: run_tests_c_linux_dbg_native_x86_default_buildonly"), but there is some ssh/rsync error when collecting the results.

**Comment by daniel-j-born** [comment]
Ping - How can I retry the tests? They're flaky and this PR has been stuck for a while because of it.

**Comment by markdroth** [comment]
It's unfortunately complicated to re-run just one test.  I've asked Kokoro to run it again.  Let's see what happens.

**Comment by daniel-j-born** [comment]
Rebased and fixed conflicts.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                FILE SIZE
 ++++++++++++++ GROWING                                                  ++++++++++++++
  +1.4%     +24 src/core/lib/channel/handshaker.cc                           +24  +1.4%
      [NEW]    +455 grpc_handshake_manager_do_handshake                         +455  [NEW]
  +0.9%     +16 src/core/ext/transport/chttp2/client/chttp2_connector.cc     +16  +0.9%
      +3.0%     +16 connected                                                    +16  +3.0%
      +0.4%      +2 on_handshake_done                                             +2  +0.4%
  +1.4%     +16 src/core/lib/http/httpcli_security_connector.cc              +16  +1.4%
      +3.2%     +16 ssl_handshake                                                +16  +3.2%
      +0.7%      +1 on_handshake_done                                             +1  +0.7%

 -+-+-+-+-+-+-+ MIXED                                                    +-+-+-+-+-+-+-
  +0.0%     +48 [None]                                                       -64  -0.0%

  +0.0%    +104 TOTAL                                                         -8  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**